src/caffe/parallel.cpp

```diff
@@ -82,14 +83,17 @@ GPUParams<Dtype>::GPUParams(shared_ptr<Solver<Dtype> > root_solver, int device)
 
   // Allocate device buffers
   CUDA_CHECK(cudaSetDevice(device));
-  CUDA_CHECK(cudaMalloc(&data_, size_ * sizeof(Dtype)));
+  buffer_device_ = device;
+  GPUMemoryManager::allocate(reinterpret_cast<void **>(&data_),
```

Comment:
Surely we don't need `reinterpret_cast` here? (Actually I'm not sure why we need a cast at all...)